roomserver/input/events.go

```diff
@@ -196,8 +197,22 @@ func processInviteEvent(
 
 	event := input.Event.Unwrap()
 
-	if err = event.SetUnsignedField("invite_room_state", input.InviteRoomState); err != nil {
-		return err
+	if len(input.InviteRoomState) > 0 {
+		// If we were supplied with some invite room state already (which is
+		// most likely to be if the event came in over federation) then use
+		// that.
+		if err = event.SetUnsignedField("invite_room_state", input.InviteRoomState); err != nil {
+			return err
+		}
+	} else {
+		// There's no invite room state, so let's have a go at building it
+		// up from local data. If we know about the room then we can insert
+		// the invite room state, if we don't then we just fail quietly.
+		if irs, ierr := buildInviteStrippedState(ctx, db, input); ierr == nil {
```

Comment:
When wouldn't we be able to build the invite room state? It seems like rarely, in which case a log line to say what went wrong would still be useful I think?

Reply:
The CS API deliberately passes `nil` here to get the roomserver to build the state up for us. I've updated the comments.